modules/sheet/tags.js

```diff
@@ -0,0 +1,217 @@
+        }
+        return result
+      } else {
+        return ids
+      }
+    })
+    var content = h('div', {
+      style: { padding: '20px' }
+    }, [
+      h('h2', {
+        style: { 'font-weight': 'normal' }
+      }, [
+        i18n('Applied Tags'),
+        h('input', {
+          type: 'search',
+          placeholder: 'filter tags',
+          'ev-input': function (ev) {
+            currentFilter.set(ev.target.value)
+          },
+          hooks: [FocusHook()],
+          style: {
+            'float': 'right',
+            'font-size': '100%'
+          }
+        })
+      ]),
+      renderTagBlock(filteredIds)
+    ])
+
+    catchLinks(content, (href, external, anchor) => {
+      if (!external) {
+        api.app.navigate(href, anchor)
+        close()
+      }
+    })
+
+    return content
+  }
+
+  function renderTagBlock (tags) {
+    var yourId = api.keys.sync.id()
+    return [
+      h('div', {
+        classList: 'TagList'
+      }, [
+        map(tags, (id) => {
+          return h('a.tag', {
+            href: `/tags/${id}`,
+            title: id
+          }, [
+            h('div.main', [
+              h('div.name', [ api.about.obs.name(id) ])
+            ]),
+            h('div.buttons', [
+              h('a.ToggleButton', {
+                'ev-click': () => {
+                  selectedTag.set(id)
+                  displayTags.set(false)
+                }
+              }, i18n('View Taggers'))
+            ])
+          ])
+        }, { idle: true, maxTime: 2 })
+      ])
+    ]
+  }
+
+  function renderTaggers (msgId, tagId) {
+    var taggerIds = api.tag.obs.messageTaggers(msgId, tagId)
+    var currentFilter = Value()
+    var taggerLookup = lookup(taggerIds, (id) => {
+      return [id, api.about.obs.name(id)]
+    })
+    var filteredIds = computed([taggerIds, taggerLookup, currentFilter], (ids, taggerLookup, filter) => {
+      if (filter) {
+        var result = []
+        for (var k in taggerLookup) {
+          if ((taggerLookup[k] && taggerLookup[k].toLowerCase().includes(filter.toLowerCase())) || k === filter) {
+            result.push(k)
+          }
+        }
+        return result
+      } else {
+        return ids
+      }
+    })
+    return h('div', {
+      style: { padding: '20px' }
+    }, [
+      h('h2', {
+        style: { 'font-weight': 'normal' }
+      }, [
+        api.about.obs.name(tagId),
+        i18n(' Taggers'),
+        h('input', {
+          type: 'search',
+          placeholder: 'filter names',
+          'ev-input': function (ev) {
+            currentFilter.set(ev.target.value)
+          },
+          hooks: [FocusHook()],
+          style: {
+            'float': 'right',
+            'font-size': '100%'
+          }
+        })
+      ]),
+      renderTaggersBlock(filteredIds)
+    ])
+  }
+
+  function renderTaggersBlock (profiles) {
+    var yourId = api.keys.sync.id()
+    profiles = api.profile.obs.rank(profiles)
+    return [
+      h('div', {
```

Comment:
`h('ProfileList',`

ProfileList might be a bit generic for mcss purposes (you don't want to collide with others). Maybe mention tags in the classname somewhere?

Reply:
I had actually used `ProfileList` so that it would pick up already existing styling. I copied the styling for now though, its duplicated but probably less likely to get accidentally broken if changes are made to one list. Maybe could be refactored out of both sheets at some point.